dot/state/storage_notify_test.go

```diff
@@ -60,6 +61,7 @@ func TestStorageState_RegisterStorageObserver_Multi(t *testing.T) {
 	ss := newTestStorageState(t)
 	ts, err := ss.TrieState(nil)
 	require.NoError(t, err)
+	ts = ts.Snapshot()
```

Comment:
why do we need to snapshot if we are overwriting?

Reply:
Because

1. `Snapshot()` was part of `TrieState()` previously
2. We don't want to modify the underlying data of the original `ts` from the storage state. We are overriding the `ts` variable, but that's just a pointer variable, and Snapshot gives it another pointer.